### PR TITLE
Poorts doorjack from TG

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Ports doorjack from TG, but without the emag nerf part. It is a very cool and usefull item to open airlocks if you don't have enough TC to by a emag.

Original PR: https://github.com/tgstation/tgstation/pull/50636

# Changelog

:cl:  MrDoomBringer, ported by SuperSlayer
rscadd: added the Airlock Authentication Override Card. Use it to open airlocks!
/:cl:

### CODE REVIEW

**Comment by 00ze-cyclone** [comment]
Maybe don’t let nuke ops and clown ops buy it since they got lot of TC so they really don’t need to save 3 on a AA card that can’t be spammed as much, it would be a noob trap for them since they need to go quickly around the station and having to wait after 3 airlock would definitely fuck them over if they relied on it.
Other than that pretty cool item.

**Comment by ghost** [comment]
> Maybe don’t let nuke ops and clown ops buy it since they got lot of TC so they really don’t need to save 3 on a AA card that can’t be spammed as much, it would be a noob trap for them since they need to go quickly around the station and having to wait after 3 airlock would definitely fuck them over if they relied on it. Other than that pretty cool item.

It is always your choice about what do you buy and what you don't. So it is okay to make ops to be able to by it.

File: code/game/objects/items/emags.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	icon_state = "emag"
+	var/type_whitelist //List of types 
+	var/charges = 3
+	var/max_charges = 3
+	var/list/charge_timers = list()
+	var/charge_time = 2 MINUTES
+
+/obj/item/card/emag/doorjack/Initialize(mapload)
+	. = ..()
+	type_whitelist = list(subtypesof(/obj/machinery/door/airlock), subtypesof(/obj/machinery/door/window)) //list of all acceptable typepaths that this device can affect
```
Comment (redmoogle) [inline code/game/objects/items/emags.dm:65]: make this a const dont do this on init

**Review by redmoogle — CHANGES_REQUESTED** [review]

File: code/game/objects/items/emags.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/emags.dm:65]: Ok

**Comment by Bibby0110** [comment]
This doesn't really fit in with the role that our current emag holds

**Pull request closed without merging**